src/component/sidebaritemgroup/index.tsx

```diff
@@ -30,10 +30,35 @@ function SidebarItemGroup(props: SidebarItemGroupProps) {
   const navigate = useNavigate();
   const { user } = useContext(UserContext);
   const userLastViewedUpdate = user?.lastUpdate || {};
+  const itemsRef = useRef<Array<HTMLDivElement | null>>([]);
+  const ventureItemRef = useRef<HTMLDivElement | null>(null);
 
-  const sortedVentureTimelines = timelines
-    .filter((t) => t.ventureId === ventureId)
-    .sort((a, b) => a.name.localeCompare(b.name));
+  const sortedVentureTimelines = useMemo(() => {
+    const result = timelines
+      .filter((t) => t.ventureId === ventureId)
+      .sort((a, b) => a.name.localeCompare(b.name));
+    itemsRef.current = itemsRef.current.slice(0, result.length);
+    return result;
+  }, [timelines, ventureId]);
+
+  useEffect(() => {
+    let el: HTMLDivElement | null = null;
+    if (currentTimeline) {
+      const index = sortedVentureTimelines.findIndex(
+        (t) => t.id === currentTimeline.id
+      );
+      if (index >= 0) {
+        el = itemsRef.current[index];
+      }
+    } else if (currentVenture?.id === ventureId) {
+      el = ventureItemRef.current;
+    }
+    if (el) {
+      el.scrollIntoView({
```

Comment:
Here's the most important part. When current timeline or venture changes, we use scrollIntoView to ensure that the current venture/timeline is visible by scrolling the sidebar scroll container up or down. This has the extra benefit of automatically scrolling to the current venture/timeline on initial render as well.